action.yaml

```diff
@@ -1,22 +1,91 @@
-name: 'Hello World' # taken from https://docs.github.com/en/actions/creating-actions/creating-a-composite-action
-description: 'Greet someone'
+name: Build and Push Image to ECR
+description: Builds and pushs an image to an AWS ECR repository
 inputs:
-  who-to-greet:  # id of input
-    description: 'Who to greet'
+  ecr-repository:
+    description: Name of ECR repository to push images to. Defaults to the Git repository's name.
+    required: false
+    default: ${{ github.repository }}
+  aws-access-key-id:
+    description: AWS secret key ID. Required.
     required: true
-    default: 'World'
-outputs:
-  random-number:
-    description: "Random number"
-    value: ${{ steps.random-number-generator.outputs.random-id }}
+  aws-secret-access-key:
+    description: AWS secret access key. Required.
+    required: true
+  aws-region:
+    description: AWS region. Required.
+    required: true
+  role-to-assume:
+    description: AWS IAM role to assume.
+    required: false
```

Comment:
If not required, it should have a default?

Reply:
I don't want to specify a value by default, so that no assume role will happen if no value is supplied. Having no default value seems to ignore the corresponding field in the action and works properly. I'm unsure if a default value of an empty string works exactly the same or not. I can't find any documentation on intended behavior, so I tried this with assumptions and it turned out okay.